tests/test_database.py

```diff
@@ -1588,7 +1588,42 @@ def test_multiplex_bad_multiplexer(self):
         self.assertEqual(str(cm.exception),
                          'expected multiplexer id 8, 16 or 24, but got 7')
 
+    def test_multiplex_dump(self):
+        filename = os.path.join('tests', 'files', 'test_multiplex_dump.dbc')
+        db = cantools.db.load_file(filename)
+        dumped_db = cantools.db.load_string(db.as_dbc_string())
+        dumped_msg = dumped_db.get_message_by_frame_id(0x100)
+
+        self.assertEqual(dumped_msg.signals[0].name, "MultiplexorSig")
+        self.assertEqual(dumped_msg.signals[0].is_multiplexer, True)
+        self.assertEqual(dumped_msg.signals[0].multiplexer_ids, None)
+        self.assertEqual(dumped_msg.signals[1].name, "MultiplexedSig")
+        self.assertEqual(dumped_msg.signals[1].is_multiplexer, False)
+        self.assertEqual(dumped_msg.signals[1].multiplexer_ids[0], 0x2A)
+        self.assertEqual(dumped_msg.signals[2].name, "UnmultiplexedSig")
+        self.assertEqual(dumped_msg.signals[2].multiplexer_ids, None)
+        self.assertEqual(dumped_msg.signals[2].is_multiplexer, False)
```

Comment:
I recommend testing like in this test: https://github.com/eerimoq/cantools/blob/master/tests/test_database.py#L1074-L1089

That is, read the DBC, check some data, and then compare `as_dbc_string()` to the original file contents.

Reply:
Another example: Vector seems to output two newlines after the VERSION"" token. Your test files have one. If you have also used CANdb++, they might have slightly changed the dbc output filter used in CANdb++...

Reply:
The goal is to output DBC-files on the same format as CANdb++. Unfortunately, I've not used that tool myself, so I don't know its DBC-file formatting. Leave your file as outputted by CANdb++, and I'll consider changing the whitespace formatting later.